src/script/miniscript.cpp

```diff
@@ -0,0 +1,295 @@
+                if (t << "s"_mst) num_s += 1;
+                args += (t << "z"_mst) ? 0 : (t << "o"_mst) ? 1 : 2;
+                acc_tl = ((acc_tl | t) & "ghij"_mst) |
+                    // Thresh contains a combination of timelocks if it has threshold > 1 and
+                    // it contains two different children that have different types of timelocks
+                    // Note how if any of the children don't have "k", the parent also does not have "k"
+                    "k"_mst.If(((acc_tl & t) << "k"_mst) && ((k <= 1) ||
+                        ((k > 1) && !(((acc_tl << "g"_mst) && (t << "h"_mst)) ||
+                        ((acc_tl << "h"_mst) && (t << "g"_mst)) ||
+                        ((acc_tl << "i"_mst) && (t << "j"_mst)) ||
+                        ((acc_tl << "j"_mst) && (t << "i"_mst))))));
+            }
+            return "Bdu"_mst |
+                   "z"_mst.If(args == 0) | // z=all z
+                   "o"_mst.If(args == 1) | // o=all z except one o
+                   "e"_mst.If(all_e && num_s == n_subs) | // e=all e and all s
+                   "m"_mst.If(all_e && all_m && num_s >= n_subs - k) | // m=all e, >=(n-k) s
+                   "s"_mst.If(num_s >= n_subs - k + 1) |  // s= >=(n-k+1) s
+                   acc_tl; // timelock info
+            }
+    }
+    assert(false);
+    return ""_mst;
+}
+
+size_t ComputeScriptLen(NodeType nodetype, Type sub0typ, size_t subsize, uint32_t k, size_t n_subs, size_t n_keys) {
+    switch (nodetype) {
+        case NodeType::PK_K: return subsize + 34;
+        case NodeType::PK_H: return subsize + 3 + 21;
+        case NodeType::OLDER: return subsize + 1 + (CScript() << k).size();
+        case NodeType::AFTER: return subsize + 1 + (CScript() << k).size();
+        case NodeType::HASH256: return subsize + 4 + 2 + 33;
+        case NodeType::HASH160: return subsize + 4 + 2 + 21;
+        case NodeType::SHA256: return subsize + 4 + 2 + 33;
+        case NodeType::RIPEMD160: return subsize + 4 + 2 + 21;
+        case NodeType::WRAP_A: return subsize + 2;
+        case NodeType::WRAP_S: return subsize + 1;
+        case NodeType::WRAP_C: return subsize + 1;
+        case NodeType::WRAP_D: return subsize + 3;
+        case NodeType::WRAP_V: return subsize + (sub0typ << "x"_mst);
+        case NodeType::WRAP_J: return subsize + 4;
+        case NodeType::WRAP_N: return subsize + 1;
+        case NodeType::JUST_1: return 1;
+        case NodeType::JUST_0: return 1;
+        case NodeType::AND_V: return subsize;
+        case NodeType::AND_B: return subsize + 1;
+        case NodeType::OR_B: return subsize + 1;
+        case NodeType::OR_D: return subsize + 3;
+        case NodeType::OR_C: return subsize + 2;
+        case NodeType::OR_I: return subsize + 3;
+        case NodeType::ANDOR: return subsize + 3;
+        case NodeType::THRESH: return subsize + n_subs + 1;
+        case NodeType::MULTI: return subsize + 3 + (n_keys > 16) + (k > 16) + 34 * n_keys;
```

Comment:
In 3a2c00d9bc3c7a34fb8727d76e9c3a691395b67c "Miniscript: type system, script creation, text notation, tests"

Several of these sizes are the same, so this could be deduplicated to be a bit easier to read.

Reply:
Also removing the `subsize` where it was not necessary (it must always be 0 for "leaf" fragments). I think it's much less confusing now.